src/Commands/BackupCommand.php

```diff
@@ -24,7 +24,13 @@ public function handle()
         try {
             $this->guardAgainstInvalidOptions();
 
-            $backupJob = BackupJobFactory::createFromArray(config('backup'));
+            $config = config('backup');
+
+            if ($this->option('connection')) {
+                array_set($config, 'backup.source.databases', $this->option('connection'));
```

Comment:
Instead of overriding the config, I'd rather see some dedicated methods on the  `BackupJobFactory` to set the databases to be backed up.

Reply:
Made `createDbDumpers` on BackupJobFactory public for that logic to be reused.